### ISSUE TITLE
Training on custom objects stuck at around 2.0 avg

### ISSUE THREAD

**Issue by JaviFuentes94** [AlexeyAB/darknet#256]
Hi,
I was trying to train YOLO with a custom dataset as explained on the repo, but I after 6000 iterations the average error is stuck at 2, and it has been there for the past 5000 iterations. 

My training set is composed of 170 images, 2 classes and around 5 objects of each class per image. While training it seems like it gets better results when the images are bigger, as the objects are bigger and YOLO has some issues with detecting small objects.  However, I am not sure how to solve this issue, or if it is even possible using this network. Some options that I was thinking about: 

- Set weight and height in the cfg file to 832 also when training. 
- Set random=0 or at least increase height and weight and reduce jitter
- Modify learning rate and continue training
- Start training again and hope it avoids this local minimum

Just to get an idea about my dataset you can see this image:

 ![test](https://user-images.githubusercontent.com/27479569/32487029-b8466154-c3a8-11e7-931e-e0c3db40560f.jpg).

 I am trying to detect both the rings and mouths of the wrenches. As you can see the resolution is not that high, but I am not sure if it can be a problem or not. 

I hope someone can help me on this one,

Javi

**Comment by AlexeyAB** [comment]
@JaviFuentes94 Hi,

I think problem in the training dataset:

* You should have ~4000 images for 2 classes (i.e. 2000 images per class), but you have only 170 images
* You should have images with resolution greater than network resolution (i.e. greater than width=416 height=416), but you have images with resolution 256x256

----

You can set random=0 and decrease jitter, and it can give your lower avg-loss on training, but will give your higher avg-loss (lower IoU) on validation dataset (on that images which isn't used in training dataset).

**Comment by JaviFuentes94** [comment]
@AlexeyAB Hi,

Thank you a lot for your input. 

I am trying to get a new set of training images with a higher resolution and I have a couple of questions. 

- Will it affect the training time if they are too big (1920x1080)?

- Do they need to be squared as the input of the network?

Thanks!

**Comment by AlexeyAB** [comment]
@JaviFuentes94 

* It isn't too big. But object size on images **preferably** should be more than ~7% = (32/416)
* Aspect ration of images should be equal in the training and validation/testing datasets. All images will be automatically resized to the network size, that usually is square.
